src/features/core/cookieConsent/ketch/browser.js

```diff
@@ -0,0 +1,83 @@
+import logger from '../../../../utils/logUtil';
+
+/* eslint-disable class-methods-use-this */
+class Ketch {
+  constructor() {
+    this.userConsentedPurposes = [];
+    this.userDeniedPurposes = [];
+
+    // updateKetchConsent callback function to update current consent purpose state
+    // this will be called from ketch ruddersatck plugin
+    window.updateKetchConsent = (consent) => {
+      if (consent.purposes) {
+        const p = consent.purposes;
+        this.userConsentedPurposes = [];
+        this.userDeniedPurposes = [];
+        Object.entries(p).forEach((e) => {
+          if (e[1]) {
+            this.userConsentedPurposes.push(e[0]);
+          } else {
+            this.userDeniedPurposes.push(e[0]);
+          }
+        });
+      }
+    };
+
+    // getKetchUserConsentedPurposes returns current ketch opted-in purposes
+    window.getKetchUserConsentedPurposes = () => this.userConsentedPurposes.slice();
+
+    // getKetchUserDeniedPurposes returns current ketch opted-out purposes
+    window.getKetchUserDeniedPurposes = () => this.userDeniedPurposes.slice();
+
+    // If ketch tag is loaded before rudderstack and has existing consent then update the consent state
+    if (window.ketchConsent) {
+      Object.entries(window.ketchConsent).forEach((e) => {
+        if (e[1]) {
+          this.userConsentedPurposes.push(e[0]);
+        } else {
+          this.userDeniedPurposes.push(e[0]);
+        }
+      });
+    }
+  }
+
+  isEnabled(destConfig) {
+    try {
+      /**
+       * Structure of ketch consent purpose destination config.
+       *
+       * "ketchConsentPurposes": [
+       * {
+       * "purpose" : "analytics"
```

Comment:
is the value "analytics" considered a unique id within the enties of the ketchConsentPurposes or the value of "analytics" can change forthe same purpose entry, like for example in different language etc?

Reply:
It is Ketch purpose code which is considered a unique id. It won't change with a different language.